### DISCUSSION TITLE
memcmp syscall

### DISCUSSION THREAD

**Discussion by pierrec** [maximecb/uvm#36] [Ideas]
Any chance to have an efficient memcmp?

**Comment by maximecb** [comment]
Hi Pierre. This can certainly be added. What is your use case?

**Comment by pierrec** [comment]
I knew you were going to ask, I should have given you the reason..
I need to compare strings and array/slice content. This can already be done but it is a bit of work to do it dierctly in UVM.
Also, it would complement the memcopy/memset syscalls already available.

**Reply by maximecb** [reply]
I've added `memcmp`, but I haven't tested it. Can you try it out and let me know if it works as you expect?

**Reply by pierrec** [reply]
A quick check validates your change :) Thanks!
